challenger.go

```diff
@@ -84,15 +88,20 @@ func NewLndChallenger(cfg *authConfig, genInvoiceReq InvoiceRequestGenerator,
 	}
 
 	invoicesMtx := &sync.Mutex{}
-	return &LndChallenger{
+	c := &LndChallenger{
```

Comment:
Need to add the `verifyInvoiceStatus` to the struct, otherwise it won't be set if it's non-nil.

Reply:
Oh wow nice catch, I think while restructuring things I completely left it out here.